lib/core_extensions.rb

```diff
@@ -51,6 +59,19 @@ def before_destroy(record)
         false
       end
     end
+
+    def check_permissions(record, klass, klass_name, what)
+      association = record.class.reflect_on_association(klass.to_sym).klass
+      association = Host if association == Host::Managed
```

Comment:
Why's this here?  It seems to skip the Authorizable check for hosts then, as it's Host::Base/Managed classes that has the module, not the Host module.